test/system/011-image.bats

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bats
+
+load helpers
+
+function setup() {
+    skip_if_remote "--sign-by does not work with podman-remote"
+
+    basic_setup
+
+    export _GNUPGHOME_TMP=$PODMAN_TMPDIR/.gnupg
+    mkdir --mode=0700 $_GNUPGHOME_TMP $PODMAN_TMPDIR/signatures
+
+    cat >$PODMAN_TMPDIR/keydetails <<EOF
+    %echo Generating a basic OpenPGP key
+    Key-Type: RSA
+    Key-Length: 2048
+    Subkey-Type: RSA
+    Subkey-Length: 2048
+    Name-Real: Foo
+    Name-Comment: Foo
+    Name-Email: foo@bar.com
+    Expire-Date: 0
+    %no-ask-passphrase
+    %no-protection
+    # Do a commit here, so that we can later print "done" :-)
+    %commit
+    %echo done
+EOF
+    GNUPGHOME=$_GNUPGHOME_TMP gpg --verbose --batch --gen-key $PODMAN_TMPDIR/keydetails
+}
+
+function check_signature() {
+    ls -laR $PODMAN_TMPDIR/signatures
+    run_podman inspect --format '{{.Digest}}' $PODMAN_TEST_IMAGE_FQN
+    local repodigest=${output/:/=}
+
+    local dir="$PODMAN_TMPDIR/signatures/libpod/${PODMAN_TEST_IMAGE_NAME}@${repodigest}"
+
+    test -d $dir || die "Missing signature directory $dir"
+    test -e "$dir/$1" || die "Missing signature file '$1'"
+
+    # Confirm good signature
+    GNUPGHOME=$_GNUPGHOME_TMP gpg --verify "$dir/$1"
```

Comment:
Any reason you don't check the actual gpg output?
```diff
index 078b422c0..57763b635 100644
--- a/test/system/011-image.bats
+++ b/test/system/011-image.bats
@@ -40,7 +40,9 @@ function check_signature() {
     test -e "$dir/$1" || die "Missing signature file '$1'"

     # Confirm good signature
-    GNUPGHOME=$_GNUPGHOME_TMP gpg --verify "$dir/$1"
+    run env GNUPGHOME=$_GNUPGHOME_TMP gpg --verify "$dir/$sigfile"
+    is "$output" ".*Good signature from .Foo.*<foo@bar.com>" \
+       "gpg --verify $sigfile"
 }


```
(Requires adding `local sigfile=$1`, which I would encourage you to do when `$1` is used in more than one or two places).